Batch of small fixes (#2247)
segmentio/segment-docs: src/connections/destinations/catalog/braze/index.md

When you Identify a user, Segment passes that user's information to Braze with `userId` as Braze's External User ID.

> If you are using a device-mode connection, Braze's SDK assigns a `device_id` and a backend identifier, `braze_id`, to every user. This allows Braze to capture anonymous activity from the device by matching on those identifiers instead of `userId`. This applies to _device-mode connections_.

Reviewer: ```suggestion
If you're using a device-mode connection, Braze's SDK assigns a `device_id` and a backend identifier, `braze_id`, to every user. This allows Braze to capture anonymous activity from the device by matching on those identifiers instead of `userId`. This applies to _device-mode connections_.
```
